### PR TITLE
[FixedPt] add the fixed-point quantize transformation and unit tests

### PR DESCRIPTION
**Pull request by SpiritSeeker**
* added the fixed-point quantize transformation, based on `fixedpt_quantize.py` from `custom/SIRA` branch.
* removed `fxpmath` dependency.
* added unit tests for op-based and dict-based quantization.

### CODE REVIEW

File: tests/transformation/test_fixedpt_quantize.py [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+    assert os.path.isfile(dl_file)
+    model = ModelWrapper(dl_file)
+    model = cleanup_model(model)
+
+    tdtype = test_details["dtype"]
+    fxp_transform = FixedPointQuantizeParams(tdtype, rounding_mode=test_details["rounding_mode"])
+    tdtype = DataType[tdtype]
+    model = model.transform(fxp_transform)
+    model = cleanup_model(model)
+    os.unlink(dl_file)
```
Comment (maltanar) [inline tests/transformation/test_fixedpt_quantize.py:479]: minor: suggest to remove downloaded files at the end of the test. in case the test fails, this helps the debug a little bit

**Review by maltanar — CHANGES_REQUESTED** [review]
Looks good overall, a few minor things could be improved - please see comments.

File: src/qonnx/transformation/fixedpt_quantize.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                    in1_t_new = self.round_func(in1_t.astype(np.float32) / self.fixedpt_dtype.scale_factor()) * self.fixedpt_dtype.scale_factor()
+                    if (in1_t_new.max() > self.fixedpt_dtype.max()) or (in1_t_new.min() < self.fixedpt_dtype.min()):
+                        warn(
+                            f"Range of {inp_name} [{in1_t_new.min():.3f}, {in1_t_new.max():.3f}] greater than"
+                            f"{self.fixedpt_dtype.get_canonical_name()} [{self.fixedpt_dtype.min():.3f}, {self.fixedpt_dtype:.max():.3f}], clipping.")
+                        in1_t_new = np.clip(in1_t_new, self.fixedpt_dtype.min(), self.fixedpt_dtype.max())
+                    model.set_initializer(inp_name, in1_t_new)
+                    model.set_tensor_datatype(inp_name, self.fixedpt_dtype)
+
+                    self.max_err[inp_name] = np.linalg.norm(in1_t.flatten() - in1_t_new.flatten(), ord=np.inf)
```
[the remark this answers is not in the document]
Comment (SpiritSeeker) [reply src/qonnx/transformation/fixedpt_quantize.py]: Incorporated this change in the new commits

File: src/qonnx/transformation/fixedpt_quantize.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+    def apply(self, model: ModelWrapper):
+        for tname, tdtype in self.fixedpt_dict.items():
+            if (in1_t := model.get_initializer(tname)) is not None:
+                if isinstance(tdtype, str):
+                    tdtype = DataType[tdtype]
+                current_dtype = model.get_tensor_datatype(tname)
+                if current_dtype == tdtype:
+                    self.max_err[tname] = 0
+                    continue
```
[the remark this answers is not in the document]
Comment (SpiritSeeker) [reply src/qonnx/transformation/fixedpt_quantize.py:67]: I don't think there would be a numerical difference. The test cases I have do not show any difference, but they might not be exhaustive. We can probably try to figure out a better solution in case any bugs arise in some corner cases.

**Review by maltanar — APPROVED** [review]
